### ISSUE TITLE
Refactor state based app data to utilize strictly API data

### ISSUE THREAD

**Issue by Oculareo** [Oculareo/REST-countries-API#15]
For development purposes, I captured the entire directory of countries and the corresponding information that we needed and then filtered and parsed from state using ContextAPI for display purposes.  After all of the main features of the app have been completed then change to API calls for different parts of the app such as searching by region or by search input values.

**Comment by Oculareo** [comment]
I did some research into different strategies to utilize on the API calls and here are the results.

The region selection box should trigger a new API call with region selection passed in resulting in a new directory of country data.

The search input can have two different approaches since dynamically calling the API on a controlled input component would result in multiple, rapid calls to the API which is not desired in this situation:

1. Have the user hit enter or click on the search icon to re-submit the API call with matching names.
2. Have the app re-submit the API call with matching names after waiting a few moments after the user finishes typing.

**Comment by Oculareo** [comment]
Another idea to tack onto this issue and future feature is to make the API call to Axios more abstract since it is something that will be highly reusable so it should be structured less for this specific project.

**Comment by Oculareo** [comment]
I created a new branch for the refactor: axios-refactor, since the main branch has an effectively functional and complete app right now.

In the new branch, I have refactored CountryDirectory to use a custom hook: useAxios which returns { response: countryData, error, loading } and the countryData is utilized exactly as the countryData retrieved previously from Context.  I also set conditionals on the URL call to axios and the API to dynamically display data matching data in the search names and selected by region.

The Country Details page needs to be updated and then the refactor will be complete.
